src/diffusers/loaders/single_file_utils.py

```diff
@@ -1859,3 +1867,195 @@ def convert_animatediff_checkpoint_to_diffusers(checkpoint, **kwargs):
             ] = v
 
     return converted_state_dict
+
+
+def convert_flux_transformer_checkpoint_to_diffusers(checkpoint, **kwargs):
+    converted_state_dict = {}
+
+    num_layers = list(set(int(k.split(".", 2)[1]) for k in checkpoint if "double_blocks." in k))[-1] + 1  # noqa: C401
+    num_single_layers = list(set(int(k.split(".", 2)[1]) for k in checkpoint if "single_blocks." in k))[-1] + 1  # noqa: C401
+    mlp_ratio = 4.0
+    inner_dim = 3072
```

Comment:
Should be okay to hardcode these no? They are the same across both models. We can grab `inner_dim` from the checkpoint, but not sure about `mlp_ratio`?

Reply:
cc: @yiyixuxu

Reply:
I think okay to hard code. I would just define them as proper constants at the top of the file.

Reply:
But if they're only applied within the scope of this function, they don't need to exist as constants that can be accessed globally?